index.js

```diff
@@ -64,46 +64,64 @@ Rollout.prototype.multi = function (keys) {
 Rollout.prototype.get = function (key, id, opt_values, multi) {
   var flags = this._handlers[key]
   var likely = this.val_to_percent(key + id)
-  var _id = {
-    id: id
-  }
-  if (!opt_values) opt_values = _id
-  if (!opt_values.id) opt_values.id = id
+  var _id = { id: id }
+  opt_values = opt_values || _id
+  opt_values.id = opt_values.id || id
   var keys = Object.keys(flags).map(function (k) {
     return key + ':' + k
   })
   var client = multi || this.client
   return new Promise(function (resolve, reject) {
     client.mget(keys, function (err, result) {
-      if (err) return reject(err)
+      if (err) {
+        return reject(err)
+      }
       resolve(result)
     })
   })
   .then(function (percentages) {
     var i = 0
     var deferreds = []
+    var output
     for (var modifier in flags) {
       // in the circumstance that the key is not found, default to original value
       if (percentages[i] === null) {
         percentages[i] = flags[modifier].percentage
       }
       if (likely < percentages[i]) {
-        if (!flags[modifier].condition) flags[modifier].condition = defaultCondition
-        var output = flags[modifier].condition(opt_values[modifier])
+        if (!flags[modifier].condition) {
+          flags[modifier].condition = defaultCondition
+        }
+        output = flags[modifier].condition(opt_values[modifier])
         if (output) {
-          if (typeof output.then === 'function') deferreds.push(output)
-          else return true
+          if (typeof output.then === 'function') {
+            // Normalize thenable to Bluebird Promise
+            // Reflect the Promise to coalesce rejections
+            deferreds.push(Promise.resolve(output).reflect())
+          } else {
+            return true
+          }
         }
       }
       i++
     }
     if (deferreds.length) {
       return Promise.all(deferreds)
       .then(function (results) {
+        var resultPromise, resultValue
         for (var i = 0, len = results.length; i < len; i++) {
-          if (results[i]) return true
+          resultPromise = results[i]
+          // Treat rejected Promises as falsey
+          if (resultPromise.isFulfilled()) {
+            resultValue = resultPromise.value()
+            // Treat resolved promises with truthy or undefined values as affirmative
+            // (`undefined` must be accepted to support `Promise.resolve()` condition)
```

Comment:
Wouldn't it be better to update the tests and make sure that the `Promise.resolve` is called with some truthy value?

Reply:
The goal of this PR was to fix the broken test without breaking backwards compatibility for existing clients.

I wanted the last version of the `0.4.x` line to have passing tests before introducing breaking API changes in #9

Reply:
From https://github.com/mix/node-rollout/pull/9#discussion_r168237592

> `false` and `null` are rejected, but arguably only `false` should be rejected because Bluebird recommends resolving Promises with `null` instead of `undefined` to silence the ["a promise was created in a handler but was not returned from it" warning](http://bluebirdjs.com/docs/warning-explanations.html#warning-a-promise-was-created-in-a-handler-but-was-not-returned-from-it)